Libraries/LibWeb/Streams/AbstractOperations.cpp

```diff
@@ -5476,43 +5477,58 @@ GC::Ref<WebIDL::Promise> transform_stream_default_sink_close_algorithm(Transform
 {
     auto& realm = stream.realm();
 
-    // 1. Let readable be stream.[[readable]].
+    // 1. Let controller be stream.[[controller]].
+    auto controller = stream.controller();
+
+    // 2. If controller.[[finishPromise]] is not undefined, return controller.[[finishPromise]].
+    if (controller->finish_promise())
+        return *controller->finish_promise();
+
+    // 3. Let readable be stream.[[readable]].
     auto readable = stream.readable();
 
-    // 2. Let controller be stream.[[controller]].
-    auto controller = stream.controller();
+    // 4. Let controller.[[finishPromise]] be a new promise.
+    controller->set_finish_promise(WebIDL::create_promise(realm));
 
-    // 3. Let flushPromise be the result of performing controller.[[flushAlgorithm]].
+    // 5. Let flushPromise be the result of performing controller.[[flushAlgorithm]].
     auto flush_promise = controller->flush_algorithm()->function()();
 
-    // 4. Perform ! TransformStreamDefaultControllerClearAlgorithms(controller).
+    // 6. Perform ! TransformStreamDefaultControllerClearAlgorithms(con`troller).
```

Comment:
typo

Reply:
how on earth did I do that :man_facepalming: